pkg/cmd/repo/rename/rename.go

```
@@ -144,51 +133,51 @@ func renameRun(opts *RenameOptions) error {
 		}
 	}
 
-	err = runRename(apiClient, input.RepoHost, input)
+	err = runRename(apiClient, oldRepo.RepoHost(), input)
 	if err != nil {
 		return fmt.Errorf("API called failed: %s, please check your parameters", err)
 	}
 
-	if !opts.flagRepo {
+	if opts.oldRepoSelector == "" {
 		cfg, err := opts.Config()
 		if err != nil {
 			return err
 		}
 
-		protocol, err := cfg.Get(input.RepoHost, "git_protocol")
+		protocol, err := cfg.Get(oldRepo.RepoHost(), "git_protocol")
 		if err != nil {
 			return err
 		}
-		remoteURL := formatRemoteURL(input, protocol)
 
-		err = git.UpdateRemote("origin", remoteURL)
+		remotes, err := opts.Remotes()
 		if err != nil {
 			return err
 		}
 
-		if opts.IO.IsStdoutTTY() {
-			fmt.Fprintf(opts.IO.Out, "%s Added remote %s\n", cs.SuccessIcon(), remoteURL)
+		baseRemote, err := remotes.FindByRepo(oldRepo.RepoOwner(), oldRepo.RepoName())
+		if err != nil {
+			return err
```

Right now, if either `opts.Remotes()` or `remotes.FindByRepo()` returns an error, that error will abort the `rename` command. I wonder whether that's necessary, and could we just ignore the error? The reasoning being that if there is no local git repo, or no git remote pointing to the old repo name, we can just proceed with the `rename` command (no reason to abort it). @samcoe thoughts?

---

I agree, the actually renaming happens at line 118 above so we have actually already renamed the repo on GH and have just failed to update the local state. I think that any error after that point should just be a warning to the user saying we were unable to update the remote due to some reason.

---

@samcoe